lib/apm-client/apm-client.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+const ElasticAPMHttpClient = require('elastic-apm-http-client')
+
+const { CENTRAL_CONFIG_OPTS } = require('../config/schema')
+const { normalize } = require('../config/config')
+const logging = require('../logging')
+
+const { NoopApmClient } = require('./noop-apm-client')
+const { getHttpClientConfig } = require('./http-apm-client')
+
+/**
+ * Returns an APM client suited for the configuration provided
+ *
+ * @param {Object} config The agent's configuration
+ * @param {Object} agent The agents instance
+ */
+function createApmClient (config, agent) {
+  let client
+  if (config.disableSend || config.contextPropagationOnly) {
+    client = new NoopApmClient()
```

Comment:
This is a slight behaviour change here that we are adding the event listeners for this code path.
I would bias towards not changing the behaviour here because this is meant to be a refactoring.
There is also no point in having the event listeners added for a NoopApmClient that we don't expect to ever emit those events.

If you agree, then the added `on ()` method to NoopApmClient could be removed. If we want NoopApmClient to mirror that part of the client API, it might be clearer to have it extend EventEmitter.

Reply:
Indeed this is a change of behaviour for the `NoopApmClient` and you're right this is beyond the scope of this PR. I'll revert this change for now.

The intent on making the clients to implement `on()` API is to enable testability of the logic that updates configuration form central configuration. Also I think future client implementations (gPRC?) must implement `on()` in order to send config back to the agent.